include/ygm/container/bag.hpp

```diff
@@ -29,6 +29,11 @@ class bag {
 
   void swap(self_type &s) { m_impl.swap(s.m_impl); }
 
+  template <typename rand_num_gen>
```

Comment:
For consistency, use the same name "RandomFunc" as: 
https://en.cppreference.com/w/cpp/algorithm/random_shuffle#Version_2